stubs/PyAutoGUI/pyautogui/__init__.pyi

```diff
@@ -1,37 +1,156 @@
 import contextlib
+from _typeshed import Incomplete
 from collections.abc import Callable, Generator, Iterable, Sequence
 from datetime import datetime
-from typing import NamedTuple, TypeVar, overload
-from typing_extensions import ParamSpec
+from typing import NamedTuple, SupportsFloat, SupportsInt, TypeVar, overload
+from typing_extensions import ParamSpec, SupportsIndex, TypeAlias
 
-# from pyscreeze import Box
+from PIL import Image
 
 class PyAutoGUIException(Exception): ...
 class FailSafeException(PyAutoGUIException): ...
 class ImageNotFoundException(PyAutoGUIException): ...
 
 _P = ParamSpec("_P")
 _R = TypeVar("_R")
+_NormalizeableXArg: TypeAlias = SupportsInt | Sequence[SupportsInt]
 
-# TODO: Complete types with pyscreeze once we can import it as a type dependency
-# Actually `pyscreeze.Box`, but typeshed doesn't currently have stubs for pyscreeze
-# (and the library doesn't have type annotations either)
+# TODO: cv2.Mat is not available as a type yet: https://github.com/microsoft/python-type-stubs/issues/211
+# cv2.Mat is just an alias for a numpy NDArray, but can't import that either.
+_Mat: TypeAlias = Incomplete
 
-class _Box(NamedTuple):
+class _Box(NamedTuple):  # Same as pyscreeze.Box
     left: int
     top: int
     width: int
     height: int
 
 def raisePyAutoGUIImageNotFoundException(wrappedFunction: Callable[_P, _R]) -> Callable[_P, _R]: ...
 
-# These functions reuse pyscreeze functions directly. See above TODO.
-def locate(*args, **kwargs) -> _Box | None: ...
-def locateAll(*args, **kwargs) -> Generator[_Box, None, None]: ...
-def locateAllOnScreen(*args, **kwargs) -> Generator[_Box, None, None]: ...
-def locateCenterOnScreen(*args, **kwargs) -> Point | None: ...
-def locateOnScreen(*args, **kwargs) -> _Box | None: ...
-def locateOnWindow(*args, **kwargs) -> _Box | None: ...
+# These functions reuse pyscreeze functions directly.
+# TODO: Once pyscreeze is typed, we can alias them to simplify this stub
+
+# _locateAll_opencv
+@overload
+def locate(
+    needleImage,
+    haystackImage,
+    grayscale: bool | None = ...,
+    limit: object | None = ...,
```

Comment:
This kwarg is always overriden. What's best to do here? `object | None`, `Any`, or just omit it?

Reply:
Answered by https://github.com/python/typeshed/pull/8652#discussion_r963151335
Just assign `object`